README.md

```diff
@@ -1,2 +1,95 @@
-# Code samples
-Code samples, mostly for my blog http://oliora.github.io.
+# Rule of Zero PIMPL
+
+A single-header C++ library for PIMPLs without having to implement any special member functions as described in [oliora's blog](http://oliora.github.io/2015/12/29/pimpl-and-rule-of-zero.html).
+
+## Code Samples
+
+Header File:
+
+```cpp
+#include "spimpl.h"
+
+class Copyable {
+public:
+  Copyable(int x);
+
+  int x();
+
+  // All five special members are compiler-generated.
+
+private:
+  class Impl;
+
+  // Movable and copyable Smart PIMPL
+  spimpl::impl_ptr<Impl> impl_;
+};
+```
+
+Implementation File:
+
+```cpp
+Copyable::Copyable(int x)
+: impl_(spimpl::make_impl<Impl>(x)) {}
+
+class Copyable::Impl {
```

Comment:
Minor issue: `Copyable::Impl` should be a struct or its member `x` should be made public, otherwise `Copyable::x()` can't access `Copyable::Impl::x`.